api/helpers/add-game-to-match.js

```diff
@@ -0,0 +1,68 @@
+module.exports = {
+  friendlyName: 'Add Game to Match',
+
+  description:
+    'Adds the provided game to the appropriate match. Will create a match if not found and will end the match if this game finishes it.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description:
+        'Game record to be added to the appropriate match. Should include an array of players',
+      required: true,
+    },
+  },
+  fn: async ({ game }, exits) => {
+    try {
+      const [player1, player2] = game.players;
+      let relevantMatch = await sails.helpers.findOrCreateCurrentMatch(player1.id, player2.id);
+      if (!relevantMatch) {
+        return exits.error('Could not add game to match');
+      }
+      if (relevantMatch.endTime || relevantMatch.winner) {
+        return exits.success(null);
+      }
```

Comment:
Does this have any sort of UX implication or way to notify the players it isn't going to count?

Reply:
I definitely want to make one, but I was thinking this could be a potential scope cut for the initial rollout. The idea being that in the beginning, we'll just tell people that ranked games now automatically track the scores and let people create ranked games and see the results of their matches in the stats page. Then we can update the GameOverDialog and some other areas to for example show you the state of your current match against your opponent and let you know the score and whether it was finished before the current game started